x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -749,7 +749,7 @@ private GetAliasesRequest createGetAliasesRequest(FieldCapabilitiesResponse resp
             } else {
                 // if the field type is the same across all this alias' indices, check the field's capabilities (searchable/aggregatable)
                 for (Entry<String, FieldCapabilities> type : types.entrySet()) {
-                    if (type.getKey() == UNMAPPED) {
+                    if (Objects.equals(type.getKey(), UNMAPPED)) {
```

Comment:
There is another similar comparison in L713, would you mind correcting that as well?

Reply:
Sure